.github/workflows/yocto-build-deploy.yml

```diff
@@ -850,41 +850,61 @@ jobs:
-      - name: Prepare artifacts
+      # https://github.com/balena-os/balena-yocto-scripts/blob/fe5debadae75e2b9cda9bd40aa2d0aced777860a/automation/include/balena-deploy.inc#L23
+      - name: Prepare hostapp artifacts
         env:
           ARTIFACTS_TAR: "${{ runner.temp }}/artifacts.tar.zst"
         run: |
-          if ! command -v zip; then
-            sudo apt-get update
-            sudo apt-get install -y zip
+          automation_dir="${{ env.automation_dir }}"
+          if [[ ! -d "$automation_dir" ]]; then
+            echo "[ERROR] automation_dir missing or not accessible"
+            exit 1
           fi
 
           source "${automation_dir}/include/balena-deploy.inc"
-          # https://github.com/balena-os/balena-yocto-scripts/blob/fe5debadae75e2b9cda9bd40aa2d0aced777860a/automation/include/balena-deploy.inc#L23
-          balena_deploy_artifacts "${SLUG}" "${DEPLOY_PATH}" true
+          balena_deploy_artifacts "${SLUG}" "${S3_DEPLOY_PATH}" false
 
-          cp -v "${WORKSPACE}/balena.yml" "${DEPLOY_PATH}/balena.yml"
+          cp -v "${WORKSPACE}/balena.yml" "${S3_DEPLOY_PATH}/balena.yml"
 
-          du -cksh "${DEPLOY_PATH}"
-          find "${DEPLOY_PATH}" -type f -exec du -h {} \;
+          find "${DEPLOY_PATH}" -exec ls -lh {} \;
 
-          tar -I zstd -cf "${ARTIFACTS_TAR}" -C "${DEPLOY_PATH}" .
-          du -h "${ARTIFACTS_TAR}"
+      # Deflate files are the main image artifact for S3 / webresource deploy
+      # is they are used by the img-maker and os download endpoints.
+      - name: Prepare deflate files
+        if: needs.balena-lib.outputs.deploy_artifact != 'docker-image'
+        env:
+          HELPER_IMAGE: balena/balena-img:6.20.26
+        run: |
+          docker run --rm \
+            -e BASE_DIR=/host/images \
+            -v "${DEPLOY_PATH}:/host/images" \
+            "${HELPER_IMAGE}" /usr/src/app/node_modules/.bin/ts-node /usr/src/app/scripts/prepare.ts
 
-      # Encrypt artifacts and remove the original tarball so it doesn't get uploaded
-      - name: Encrypt artifacts
+          find "${DEPLOY_PATH}" -exec ls -lh {} \;
+
+      # Encrypt .img and .img.zip files and remove the originals
+      # Encryption is required for private device types, and signed images.
+      - name: Encrypt signed/private artifacts
         if: inputs.sign-image || needs.balena-lib.outputs.is_private == 'true'
         env:
-          ARTIFACTS_TAR: "${{ runner.temp }}/artifacts.tar.zst"
-          ARTIFACTS_ENC: "${{ runner.temp }}/artifacts.tar.zst.enc"
           PBDKF2_PASSPHRASE: ${{ secrets.PBDKF2_PASSPHRASE }}
         run: |
-          openssl enc -v -e -aes-256-cbc -k "${PBDKF2_PASSPHRASE}" -pbkdf2 -iter 310000 -md sha256 -salt -in "${ARTIFACTS_TAR}" -out "${ARTIFACTS_ENC}"
-          rm "${ARTIFACTS_TAR}"
+          set -x
+          for file in "${S3_DEPLOY_PATH}"/**/*.img "${S3_DEPLOY_PATH}"/**/*.img.zip "${S3_DEPLOY_PATH}"/**/*.docker "${S3_DEPLOY_PATH}"/**/*.deflate; do
```

Comment:
extreme nit - but is there anything that is not the images / hostapp files that should be encrypted? I don;t think there is, just sanity checking - the manifest, kernel module headers etc?